### PR TITLE
Rolling statistics

### LINKED ISSUE #1480: Add a rolling_statistics
### Addressing a Problem?

We already have `statistics` and `thresholded_statistics`, but a `rolling_statistics` would be very useful for anything related to hydrology and probably other domains too.

Use cases include for example computing the `maximum 14-day mean flow (q14max)`, or the `minimum 7-day mean flow (q7min)`.

### Potential Solution

Possible code to be used:
```
def roll_stats(da: xr.DataArray, window: int, window_reducer: str, reducer: str, freq: str, **indexer) -> xr.DataArray:
    """Calculate a stat on a rolling stat."""
    rolled = getattr(da.rolling(time=window, center=True), window_reducer)()
    rolled_indexed = select_time(rolled, **indexer)
    out = getattr(rolled_indexed.resample(time=freq), reducer)()
    out.attrs["units"] = da.attrs["units"]  # on peut faire mieux
    return out
```

### Additional context

_No response_

### Contribution

- [X] I would be willing/able to open a Pull Request to contribute this feature.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by RondeauG**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #1480
- [x] Tests for the changes have been added (for bug fixes / features)
  - [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] CHANGES.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Adds `xclim.indices.select_rolling_resample_op` that acts in a way similar to `select_resample_op`, but with a rolling window beforehand.

### CODE REVIEW

**Comment by RondeauG** [comment]
I've done it as a separate index for now, but we could also just add the `window` argumwents directly in `select_resample_op` and disable them by default.

File: xclim/indices/generic.py [hunk 1]
```diff
@@ -104,6 +104,58 @@ def select_resample_op(
...
+    r = rolled.resample(time=freq)
+    if isinstance(op, str):
+        out = getattr(r, op.replace("integral", "sum"))(dim="time", keep_attrs=True)
+    else:
+        with xr.set_options(keep_attrs=True):
+            out = r.map(op)
+        op = op.__name__
+    if out_units is not None:
+        return out.assign_attrs(units=out_units)
+    return to_agg_units(out, da, op)
```
[the remark this answers is not in the document]
Comment (RondeauG) [reply xclim/indices/generic.py:156]: `argmax/min` on `window_op` doesn't really make sense, or at least I can't think of a use case right now. I'm fine if that's not supported.

File: tests/test_generic.py [hunk 2]
```diff
@@ -31,6 +31,41 @@ def test_season(self, q_series):
...
+
+    def test_freq(self, q_series):
+        q = q_series(np.arange(1, 366 + 365 + 365 + 1))  # 1st year is leap
+        o = generic.select_rolling_resample_op(
+            q, "max", window=3, window_center=True, window_op="sum", freq="MS"
+        )
+        np.testing.assert_array_equal(
+            [np.sum([30, 31, 32]), np.sum([30 + 29, 31 + 29, 32 + 29])],
+            o.isel(time=slice(0, 2)).values,
+        )
```
Comment (aulemahal) [inline tests/test_generic.py:66]: Could you add an assertion to ensure the output units are what you expected ?

Original q is in "m3 s-1", after the rolling sum it is now "m3" and then after the max it should still be "m3".

I think this doesn't work. `to_agg_units` doesn't cover the case where the original units have a "temporal" dimension before a sum.

**Comment by aulemahal** [comment]
I added the last `assert`on the third test. This now requires #1649 to pass.

**Review by aulemahal — APPROVED** [review]
My approval.

**Pull request merged**